src/ceph-disk/ceph_disk/main.py

```diff
@@ -2545,6 +2567,60 @@ def get_dmcrypt_type(args):
             return None
 
 
+class Secrets(object):
+
+    def __init__(self):
+        secret, stderr, ret = command(['ceph-authtool', '--gen-print-key'])
+        LOG.debug("stderr " + stderr)
+        assert ret == 0
```

Comment:
`command_check_call` does capture the output:

    try:
        return subprocess.check_call(arguments)
    except subprocess.CalledProcessError as error:
        if exit:
            if error.output:
                LOG.error(error.output)
            raise SystemExit(
                "'{cmd}' failed with status code {returncode}".format(
                    cmd=command,
                    returncode=error.returncode,
                )
            )
        raise

See the `error.output` usage? This should capture your *'end of the world'* situations you described, like `ceph-authtool` not installed:

    In [1]: from ceph_disk.main import command_check_call

    In [2]: command_check_call(['ceph-authtool', '-v'])
    ---------------------------------------------------------------------------
    ExecutableNotFound                        Traceback (most recent call last)
    <ipython-input-2-b676651fa3f8> in <module>()
    ----> 1 command_check_call(['ceph-authtool', '-v'])

    /Users/alfredo/python/upstream/ceph/src/ceph-disk/ceph_disk/main.py in command_check_call(arguments, exit)
        529     otherwise.
        530     """
    --> 531     arguments = _get_command_executable(arguments)
        532     command = ' '.join(arguments)
        533     LOG.info('Running command: %s', command)

    /Users/alfredo/python/upstream/ceph/src/ceph-disk/ceph_disk/main.py in _get_command_executable(arguments)
        445         command_msg = 'Could not run command: %s' % ' '.join(arguments)
        446         executable_msg = '%s not in path.' % arguments[0]
    --> 447         raise ExecutableNotFound('%s %s' % (executable_msg, command_msg))
        448
        449     # swap the old executable for the new one

    ExecutableNotFound: ceph-authtool not in path. Could not run command: ceph-authtool -v

I find that the error handling has a good error report, that says exactly what happened, as opposed to an `AssertionError`. Sensible error messages aren't pointless, but maybe I am not following in what context saying "command failed: $command,  with error code $code, and with $output" would not be useful? 

This `assert` usage is almost doing the same, which is why I am saying it should just use the helpers that exist already.